File: wrong_code/ValueCheck.py
import pandas as pd
import re
def value(cell_value):
        # 返回值修改成了如果是单独的整数型就返回整数型，否则返回字符串
        # 比如： 字段是 2007 就直接返回 int 型，字段是 1011;1012;1013 就还是返回原类型
        if cell_value is not None and cell_value != "" and cell_value != " ":
            try:
                cell_value = str(cell_value)
                number_pattern = r'^\d+(\.\d+)?$'#correct: number_pattern = r'^-?\d+(\.\d+)?$'
                if re.match(number_pattern, cell_value):
                    # 检查是否可以转换为浮点数
                    float_value = float(cell_value)
                    int_value = int(float_value)
                    # 检查小数部分是否为零
                    if float_value - int_value == 0:
                        return int_value,type(int_value)
                    else:
                        #wrong
                        return float_value,type(cell_value) #correct: return cell_value,type(cell_value)
                else:
                    return cell_value,type(cell_value)
            except ValueError:
                return cell_value,type(cell_value)
        else:
            return None,type(cell_value)

if __name__ == "__main__":
    print(value(input("请输入一个值：")))
    #print(value())

File: hardToTest/ValueCheck/ValueCheck.java
import java.util.Scanner;
import java.util.regex.Pattern;
import java.util.regex.Matcher;

public class ValueCheck {

    // 定义正则表达式
    private static final String NUMBER_PATTERN = "^-?\\d+(\\.\\d+)?$";

    // 实现 value 方法
    public static Object[] value(String cellValue) {
        // 如果 cellValue 为 null 或为空白
        if (cellValue == null || cellValue.trim().isEmpty()) {
            return new Object[]{null, String.class};
        }

        try {
            // 转为字符串处理
            cellValue = cellValue.trim();
            Pattern pattern = Pattern.compile(NUMBER_PATTERN);
            Matcher matcher = pattern.matcher(cellValue);

            if (matcher.matches()) {
                // 如果是数字，尝试解析为浮点数和整数
                double floatValue = Double.parseDouble(cellValue);
                int intValue = (int) floatValue;

                // 检查小数部分是否为零
                if (floatValue - intValue == 0) {
                    return new Object[]{intValue, Integer.class};
                } else {
                    return new Object[]{cellValue, String.class};
                }
            } else {
                // 非数字，返回字符串
                return new Object[]{cellValue, String.class};
            }
        } catch (NumberFormatException e) {
            // 异常时，返回原值和字符串类型
            return new Object[]{cellValue, String.class};
        }
    }

    // 主方法，用于测试 value 函数
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("请输入一个值：");
        String input = scanner.nextLine();

        Object[] result = value(input);
        System.out.println("返回值: " + result[0]);
        System.out.println("返回类型: " + ((Class<?>) result[1]).getName());

        scanner.close();
    }
}
